ale_linters/perl/perl.vim

```diff
@@ -27,12 +27,20 @@ function! ale_linters#perl#perl#Handle(buffer, lines) abort
     let l:output = []
     let l:basename = expand('#' . a:buffer . ':t')
 
+    let type = 'E'
+    let syntax_ok = matchstr(a:lines[-1], 'syntax OK')
+    if syntax_ok != ''
+        let type = 'W'
+    endif
+
+    let seen = {}
```

Comment:
You should install `Vader` for checking your Vim code. Check out the output from the Travis CI build for the warnings it will give you. It will complain about your variable names.